File: Jan9/ModExp.java
import java.math.BigInteger;
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;

class ModExp
{
	public static void main(String[] args) throws IOException
	{
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

		int numCases = Integer.parseInt(br.readLine());

		for (int i = 1; i <= numCases; i++)
		{
			String line = br.readLine();
			String[] lineArr = line.split(" ");

			BigInteger b = new BigInteger(lineArr[0]);
			BigInteger e = new BigInteger(lineArr[1]);
			BigInteger m = new BigInteger(lineArr[2]);

			System.out.println(i + ". " + b.modPow(e, m));
		}
	}
}

File: Jan9/modexp.py
numCases = int(raw_input())

for i in xrange(1, numCases+1):
	line_arr = raw_input().split()
	b = long(line_arr[0])
	e = long(line_arr[1])
	m = long(line_arr[2])

	c = pow(b, e, m)

	print "%d. %d" % (i , c)